### PR TITLE
fix(openai): TypeError: '<' not supported between instances of 'NoneType' and 'int' in embeddings_wrappers.py

### LINKED ISSUE #1044: TypeError: '<' not supported between instances of 'NoneType' and 'int'
Sentry Issue: [OPENLLMETRY-K](https://traceloop.sentry.io/issues/5321245019/?referrer=github_integration)

```
TypeError: '<' not supported between instances of 'NoneType' and 'int'
  File "/home/martin/.poetry/envs/src-GLSgO57w-py3.10/lib/python3.10/site-packages/opentelemetry/instrumentation/openai/utils.py", line 124, in wrapper
    return func(*args, **kwargs)
  File "/home/martin/.poetry/envs/src-GLSgO57w-py3.10/lib/python3.10/site-packages/opentelemetry/instrumentation/openai/shared/embeddings_wrappers.py", line 171, in _handle_response
    _set_embeddings_metrics(
  File "/home/martin/.poetry/envs/src-GLSgO57w-py3.10/lib/python3.10/site-packages/opentelemetry/instrumentation/openai/shared/embeddings_wrappers.py", line 205, in _set_embeddings_metrics
    token_counter.add(val, attributes=attributes_with_token_type)
  File "/home/martin/.poetry/envs/src-GLSgO57w-py3.10/lib/python3.10/site-packages/opentelemetry/metrics/_internal/instrument.py", line 206, in add
    self._real_instrument.add(amount, attributes)
  File "/home/martin/.poetry/envs/src-GLSgO57w-py3.10/lib/python3.10/site-packages/opentelemetry/sdk/metrics/_internal/instrument.py", line 154, in add
    if amount < 0:
```

### PR DESCRIPTION
**Pull request by kowshik24**
This pull request fixes the issue where a TypeError occurs due to a comparison between NoneType and int The error was happening in the `_set_embeddings_metrics `function when recording token counts. The fix ensures that `None `values are handled appropriately by logging an error and skipping the recording.

Issue Reference: Fixes #1044

Changes Made:

Added a check to handle `None `values in the `usage `dictionary.
Logged an error message when a `None ` value is encountered.

### CODE REVIEW

**Review by nirga — COMMENTED** [review]
Thanks @kowshik24! Mind adding a test for this?

**Comment by kowshik24** [comment]
@nirga yeah sure! on it.

File: packages/opentelemetry-instrumentation-openai/tests/traces/test_embedding_metrics_hander.py
Comment (nirga) [inline packages/opentelemetry-instrumentation-openai/tests/traces/test_embedding_metrics_hander.py:1]: nit: handler

**Review by nirga — APPROVED** [review]
Nice @kowshik24! Left a small comment

File: packages/opentelemetry-instrumentation-openai/tests/traces/test_embedding_metrics_hander.py
Comment (kowshik24) [reply packages/opentelemetry-instrumentation-openai/tests/traces/test_embedding_metrics_hander.py:1]: @nirga updated

**Review by nirga — COMMENTED** [review]
Thanks @kowshik24, mind rebasing and fixing the lint errors?

**Review by nirga — APPROVED** [review]

**Comment by nirga** [comment]
Thanks @kowshik24! Great work :)

**Comment by nirga** [comment]
@kowshik24 can you look into the lint issues?

**Comment by kowshik24** [comment]
@nirga linting done! Can you please check?

**Comment by nirga** [comment]
@kowshik24 still seems to fail. Make sure to run locally `nx run-many -t lint`

**Pull request merged**